### ISSUE TITLE
Typescript: never as type parameter causes error: unknown type: undefined

### ISSUE THREAD

**Issue by hccampos** [prettier/prettier#2718]
When using the `never` type as a type parameter, prettier throws an `unknown type: undefined` error. This happens, for example, when you have code like:

```
Observable.empty<never>();
```

The full prettier stack trace for the error:
```
xxx.ts: Error: unknown type: undefined
    at genericPrintNoParens (/usr/lib/node_modules/prettier/index.js:7845:13)
    at genericPrint (/usr/lib/node_modules/prettier/index.js:5288:55)
    at comments$3.printComments.p (/usr/lib/node_modules/prettier/index.js:9929:14)
    at Object.printComments (/usr/lib/node_modules/prettier/index.js:1771:19)
    at printGenerically (/usr/lib/node_modules/prettier/index.js:9927:24)
    at FastPath$1.map (/usr/lib/node_modules/prettier/index.js:1979:19)
    at printTypeParameters (/usr/lib/node_modules/prettier/index.js:8510:48)
    at genericPrintNoParens (/usr/lib/node_modules/prettier/index.js:7445:14)
    at genericPrint (/usr/lib/node_modules/prettier/index.js:5288:55)
    at comments$3.printComments.p (/usr/lib/node_modules/prettier/index.js:9929:14)
```

Prettier *1.6.1*

**Comment by azz** [comment]
Thanks for the report - I've fixed this in our dependency and we can close this when 
https://github.com/eslint/typescript-eslint-parser/pull/381 is merged.

**Comment by JamesHenry** [comment]
The fix has been merged into the ts-2.5 branch of typescript-eslint-parser. I am going to also merge ts-2.5 into master now that 2.5 is the stable version of TypeScript, so up to you @azz which commit hash you want to grab to fix this issue.

**Comment by JamesHenry** [comment]
Actually, don't worry, I'll submit the PR myself in a sec
